storage/remote/generic/generic.go

```diff
@@ -30,9 +30,20 @@ type Client struct {
 }
 
 // NewClient creates a new Client.
-func NewClient(address string, timeout time.Duration) *Client {
-	// TODO: Do something with this error.
-	conn, err := grpc.Dial(address, grpc.WithInsecure(), grpc.WithTimeout(timeout))
+func NewClient(address string, compression string, timeout time.Duration) *Client {
+	var cp grpc.Compressor
+	switch compression {
+	case "gzip":
+		cp = grpc.NewGZIPCompressor()
+	case "snappy":
+		cp = &snappyCompressor{}
+	case "":
+		// No compression.
+	default:
+		log.Fatalf("Invalid generic remote compression type %q", compression)
```

Comment:
Agreed - I followed the existing scheme that we had for the other cases here, but it's probably best to propagate errors all the way back to main and then fatal there. Then we can do the same with the InfluxDB client (before removing it), where we also currently have a fatal log: https://github.com/prometheus/prometheus/blob/master/storage/remote/influxdb/client.go#L39